### PR TITLE
Run uploads in submodules, create empty language files & upgrade all packages

### PR DESCRIPTION
**Pull request by Vainock**
### Description
- Increase reliability
- Upgrade all packages for security
- Make it possible to run the action in submodules (obs-browser, obs-websocket, ...)
- Create empty language files for languages like en-GB to no longer make the log show warnings for missing language files
- Add missing JSDoc
- Remove handling for enc-amf
- Generate reports simultaneously because [reasons](https://community.crowdin.com/t/report-generation-via-api-fails-due-to-conflict/4643)
- Refactors

I wasn't happy with https://github.com/obsproject/obs-crowdin-sync/pull/12 and closed it. After force-pushing to the branch again I wasn't allowed to re-open it.

### Motivation and Context
Security & reliability are important.

### How Has This Been Tested?
By testing all scenarios I could come up with in a forked org on GitHub and a private Crowdin project + unit tests.

### Types of changes

- Tweak (non-breaking change to improve existing functionality)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

**Comment by RytoEX** [comment]
Aside from the merge conflict due to #15, what's the blocker here? Is there any large risk of just merging this once the merge conflict is resolved? I'm not a huge fan of blocking dependency updates on a review of a larger rework.

File: src/download.ts [hunk 1]
```diff
@@ -184,36 +154,27 @@ export async function getTranslators(targetLanguageIds: string[]): Promise<strin
...
-		);
-	}
+		const { status: reportStatus, identifier: reportId } = (
+			await reportsApi.generateReport(PROJECT_ID, {
+				name: 'top-members',
+				schema: {
+					unit: 'words',
+					format: 'json',
+					dateFrom: '2014-01-01T00:00:00+00:00',
+					dateTo: '2030-01-01T00:00:00+00:00',
```
Comment (WizardCM) [inline src/download.ts:166]: Are we comfortable hard-coding an end date? What if we used either `Date` or `dayjs` to set a date x years in the future?

File: .eslintrc [hunk 2]
```diff
@@ -4,18 +4,16 @@
 		"es2020": true,
 		"jest": true
 	},
-	"extends": ["plugin:jsdoc/recommended", "plugin:prettier/recommended"],
```
Comment (WizardCM) [inline .eslintrc:7]: As we're using typescript, I'm OK removing JSDoc in theory (as TS will cover most bits), but I'd rather keep some form of required function description linting.

File: src/download.ts [hunk 3]
```diff
@@ -20,54 +20,36 @@ const { reportsApi, translationsApi, usersApi, projectsGroupsApi, sourceFilesApi
...
-		) {
+		if (await FSE.pathExists(`${submodulePath}/data/locale/en-US.ini`)) {
 			submodules.push(submodulePath.split('/')[1]);
 		}
 	}
 	return submodules;
 }
 
-/**
- * Remove all translations to prevent keeping dropped languages.
```
Comment (WizardCM) [inline src/download.ts:58]: Please don't delete function descriptions.

File: src/download.ts [hunk 4]
```diff
@@ -93,8 +75,6 @@ function getDetachedSubmodules(submodules: string[]): string[] {
 }
 
 /**
- * Mappes file ids to their export paths.
- *
  * @returns File ids mapped to their export path.
```
Comment (WizardCM) [inline src/download.ts:78]: Changes here are inconsistent throughout the PR - the description is removed, but the return description is kept. Please keep both.

**Review by WizardCM — CHANGES_REQUESTED** [review]
Functionally, the code looks even better than before. Just a couple minor notes.

File: .eslintrc (hunk 2 above)
Comment (Vainock) [reply .eslintrc]: I'm fine with reverting this removal.

**Comment by Vainock** [comment]
@WizardCM Done.

**Comment by Vainock** [comment]
Thank you. Merging this unlocked https://github.com/obsproject/obs-studio/pull/8773.

**Pull request merged**